adagucserverEC/CImgRenderPolylines.cpp

```diff
@@ -143,29 +193,56 @@ void CImgRenderPolylines::render(CImageWarper *imageWarper, CDataSource *dataSou
               cnt++;
             } else {
               CDBDebug("status: %d %d [%f,%f]", status, j, tprojectedX, tprojectedY);
-              //                         dlat=CCONVERTUGRIDMESH_NODATA;
-              //                         dlon=CCONVERTUGRIDMESH_NODATA;
             }
-            //                       projectedX[j]=dlon;
-            //                       projectedY[j]=height-dlat;
           }
 
-          //                    CDBDebug("Draw polygon: %d points (%d)", cnt, numPoints);
+          // CDBDebug("Draw polygon: %d points (%d)", cnt, numPoints);
           drawImage->poly(projectedX, projectedY, cnt, drawPointLineWidth, drawPointLineColor2, drawPointLineColor2, true, false);
-          //                    break;
+          // Determine centroid for first polygon.
+          if (firstPolygon) {
+            Point2D centroid = getCentroid(polyX, polyY, numPoints);
+            double centroidX = centroid.x;
+            double centroidY = centroid.y;
+            int status = 0;
+            if (projectionRequired) status = imageWarper->reprojfromLatLon(centroidX, centroidY);
+            if (!status) {
+              int dlon, dlat;
+              dlon = int((centroidX - offsetX) / cellSizeX) + 1;
+              dlat = int((centroidY - offsetY) / cellSizeY);
+              std::map<std::string, FeatureProperty *>::iterator it;
+              CT::string featureId;
+              Feature *f2 = feature;
+              std::map<std::string, FeatureProperty *> *fp = f2->getFp();
+              std::map<std::string, FeatureProperty *> *featurePropertyMap = feature->getFp();
+              it = feature->getFp()->find(std::string(featureStyle.propertyName.c_str()));
+              if (it != feature->getFp()->end()) {
+                featureId.print(it->second->toString(featureStyle.propertyFormat));
+              } else {
+                featureId = feature->getId();
+              }
+              CColor labelColor(featureStyle.fontColor);
+              drawImage->drawCenteredTextNoOverlap(dlon, height - dlat, featureStyle.fontFile.c_str(), featureStyle.fontSize, featureStyle.angle, featureStyle.padding, featureId, labelColor,
+                                                   noOverlap, rects);
+            } else {
+              CDBDebug("Status: %d centroid [%f,%f]", status, centroidX, centroidY);
```

Comment:
Remove or make it depend on a macro setting

Reply:
Done